File: formula-cs/Formula/TokenTree/RootNode.cs
namespace Formula.TokenTree;

internal class RootNode<T> : Node<T> {

    public void Walk(string text, int startIndex, List<TokenMatch<T>> matches) {
        Walk(text, startIndex, startIndex, matches);
    }

    public override void Walk(string text, int startIndex, int currentIndex, List<TokenMatch<T>> matches)
    {
        WalkChildren(text, startIndex, currentIndex, matches);
    }

    public override bool Equals(object? obj)
    {
        return false;
    }

    public override int GetHashCode()
    {
        return 0;
    }
}

File: formula-java/src/main/java/org/formula/parse/tree/RootNode.java
package org.formula.parse.tree;

import java.util.List;

public class RootNode<T> extends Node<T> {

    public void walk(String text, int startIndex, List<TokenMatch<T>> matches) {
        walk(text, startIndex, startIndex, matches);
    }

    @Override
    public void walk(String text, int startIndex, int currentIndex, List<TokenMatch<T>> matches) {
        walkChildren(text, startIndex, currentIndex, matches);
    }

    @Override
    public boolean equals(Object other) {
        return false;
    }
}